sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Listeners/ServiceBusListener.cs

```diff
@@ -70,12 +70,38 @@ internal sealed class ServiceBusListener : IListener, IScaleMonitorProvider
             _loggerFactory = loggerFactory;
             _logger = loggerFactory.CreateLogger<ServiceBusListener>();
 
-            _client = new Lazy<ServiceBusClient>(() => clientFactory.CreateClientFromSetting(connection));
-            _batchReceiver = new Lazy<ServiceBusReceiver>(() => _messagingProvider.CreateBatchMessageReceiver(_client.Value, _entityPath));
-            _messageProcessor = new Lazy<MessageProcessor>(() => _messagingProvider.CreateMessageProcessor(_client.Value, _entityPath));
-            _sessionMessageProcessor = new Lazy<SessionMessageProcessor>(() => _messagingProvider.CreateSessionMessageProcessor(_client.Value, _entityPath));
+            _client = new Lazy<ServiceBusClient>(
+                () =>
+                    clientFactory.CreateClientFromSetting(connection));
+
+            _batchReceiver = new Lazy<ServiceBusReceiver>(
+                () => _messagingProvider.CreateBatchMessageReceiver(
+                    _client.Value,
+                    _entityPath,
+                    options.ToReceiverOptions()));
+
+            _messageProcessor = new Lazy<MessageProcessor>(
+                () => _messagingProvider.CreateMessageProcessor(
+                    _client.Value,
+                    _entityPath,
+                    options.ToProcessorOptions()));
+
+            _sessionMessageProcessor = new Lazy<SessionMessageProcessor>(
+                () => _messagingProvider.CreateSessionMessageProcessor(
+                    _client.Value,
+                    _entityPath,
+                    options.ToSessionProcessorOptions()));
```

Comment:
This would be updated to take the AutoComplete property from ServiceBusTriggerAttribute into account.